test/client/fixtures/page-navigation-watch-test.js

```diff
@@ -26,328 +27,209 @@ QUnit.testDone(function () {
+        .then(function (url) {
+            strictEqual(url, iframeLocation + 'index.html');
         });
 });
 
-asyncTest('window.location = ...', function () {
-    navigateIframe('window.location = "./index.html";')
-        .then(function (e) {
-            strictEqual(e, iframeLocation + 'index.html');
-            start();
-        })
-        .catch(function () {
-            ok(false, 'timeout exceeded');
-            start();
+test('window.location = ...', function () {
+    return navigateIframe('window.location = "./index.html";')
+        .then(function (url) {
+            strictEqual(url, iframeLocation + 'index.html');
         });
 });
 
-asyncTest('location.assing(...)', function () {
-    navigateIframe('location.assign("./index.html");')
-        .then(function (e) {
-            strictEqual(e, iframeLocation + 'index.html');
-            start();
-        })
-        .catch(function () {
-            ok(false, 'timeout exceeded');
-            start();
+test('location.assing(...)', function () {
+    return navigateIframe('location.assign("./index.html");')
+        .then(function (url) {
+            strictEqual(url, iframeLocation + 'index.html');
         });
 });
 
-asyncTest('location.replace(...)', function () {
-    navigateIframe('location.replace("./index.html");')
-        .then(function (e) {
-            strictEqual(e, iframeLocation + 'index.html');
-            start();
-        })
-        .catch(function () {
-            ok(false, 'timeout exceeded');
-            start();
+test('location.replace(...)', function () {
+    return navigateIframe('location.replace("./index.html");')
+        .then(function (url) {
+            strictEqual(url, iframeLocation + 'index.html');
         });
 });
 
-asyncTest('location.reload(...)', function () {
-    navigateIframe('location.reload();')
-        .then(function (e) {
-            strictEqual(e, iframeLocation);
-            start();
-        })
-        .catch(function () {
-            ok(false, 'timeout exceeded');
-            start();
-        });
-});
-
-asyncTest('Hash via location.href', function () {
-    navigateIframe('location.href += "#hash";')
-        .then(function () {
-            ok(false);
-            start();
-        })
-        .catch(function () {
-            ok(true);
-            start();
+test('location.reload(...)', function () {
+    return navigateIframe('location.reload();')
+        .then(function (url) {
+            strictEqual(url, iframeLocation);
         });
 });
 
-asyncTest('Hash via window.location (GH-917)', function () {
-    navigateIframe('window.location = "#hash";')
-        .then(function () {
-            ok(false);
-            start();
-        })
-        .catch(function () {
-            ok(true);
-            start();
+test('Hash via location.href', function () {
+    return navigateIframe('location.href += "#hash";', { timeout: 500 })
+        .catch(function (reason) {
```

Comment:
we should check that the `pageNavigationTriggered` event is not triggered here. The previous version of the test checked it. But the modified don't do it.
Now if the event will be raised the `navigateIframe` promise will be resolved and test will finish succesfully but it should fail.